### ISSUE TITLE
Console Ninja remains inactive and doesn't detect the live server

### ISSUE THREAD

**Issue by smcenlly** [wallabyjs/console-ninja#183]
Dear Console Ninja developers, 

I'm trying to use your extension with my JavaScript project that runs on VS Code's live server plugin. However, Console Ninja remains inactive and doesn't detect the live server. 

I've already tried:

- Enabling live server in extension settings
- Restarting VS Code

I'm attaching the logs that contain the following:

- Console Ninja tries starting and stopping host 
- It installs necessary files
- But doesn't find any tools to patch
- And remains inactive

Could you please check if there is some problem with detecting live server or integration with it? Maybe I'm missing some step in the setup.

It would be great if you could advise on getting Console Ninja to work with live server, as it seems a very useful extension for debugging.

Thank you for your time!
Any help is much appreciated.

Let me know if you need any other details from my end.

![image](https://github.com/wallabyjs/console-ninja/assets/8464662/876fe289-b74d-4eac-b033-97e9cb4d342c)

i use live server 
![image](https://github.com/wallabyjs/console-ninja/assets/8464662/cb97366f-0d79-47ea-b474-d6afc8bb3dcf)

and latest node
![image](https://github.com/wallabyjs/console-ninja/assets/8464662/2ac1b6cc-125e-49de-9975-9d15e2e32c92)

could you help me , please?

_Originally posted by @Alecto in https://github.com/wallabyjs/console-ninja/issues/120#issuecomment-1665985791_

**Comment by ArtemGovorov** [comment]
@Alecto Can you please do the following:
- open Ninja `trace.log` file, clear and save it,
- double check that you have Live Server extension checked in the Ninja VS Code settings,
- close all instances of VS Code, open an instance of VS Code with your project,
- start Live Server on the project,
- copy and content of the Ninja `trace.log` and paste it here in response.

Please also share the following:
- What is the version of the Live Server extension that you have installed?

**Comment by Alecto** [comment]
thank you for your help, I did all the steps according to the instructions


> double check that you have Live Server extension checked in the Ninja VS Code settings,

![image](https://github.com/wallabyjs/console-ninja/assets/8464662/49fbe723-c96d-4b06-9140-8300f585e9a8)


> close all instances of VS Code, open an instance of VS Code with your project,
+
i work with a single instance of the vscode


> start Live Server on the project,
+
![image](https://github.com/wallabyjs/console-ninja/assets/8464662/0596505a-dc72-4a43-9554-24918f6e5a18)


> What is the version of the Live Server extension that you have installed?
latest v5.7.9
![image](https://github.com/wallabyjs/console-ninja/assets/8464662/f766ba62-6783-4551-8d32-e6692e77b6da)


```

08:21:49.041 info extension stopping host
08:21:49.298 info extension host process output: port:[59548]
08:21:49.279 info host ------------ starting ------------
08:21:49.293 info host removing cache folder: f5b0a8670fbdee88709b4ceefa2b7f9c
08:21:49.298 info extension 
08:21:49.308 info host client connected: extension { address: '::', family: 'IPv6', port: 59548 }
08:21:49.395 info host starting host: {"extensionFolder":"d:\\IDE\\VSCodeForStream\\data\\extensions\\wallabyjs.console-ninja-0.0.192","runtimeHookFile":"d:\\IDE\\VSCodeForStream\\data\\extensions\\wallabyjs.console-ninja-0.0.192\\out\\runtimeHook\\tracer.js","globalErrorHandlerFile":"d:\\IDE\\VSCodeForStream\\data\\extensions\\wallabyjs.console-ninja-0.0.192\\out\\runtimeHook\\errorHandler.js","filesToInstrument":[{"path":"d:\\webprojects\\learning-templates\\_____class_____\\___javascript-start\\main.js","reason":"search"},{"path":"d:\\webprojects\\learning-templates\\_____class_____\\___javascript-start\\assets\\jasmine\\main.3.spec.js","reason":"search"},{"path":"d:\\webProjects\\learning-templates\\_____class_____\\___javascript-start\\index.html","reason":"open"}],"debug":false,"config":{"dateTimeFormat":"HH:mm:ss.SSS","showDateTime":true,"installBinToPath":true,"openLinksInEditor":true,"maxLogViewerEntries":15000,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":false,"searchUrl":"https://www.google.com/search?q={query}","enableProFeatures":true,"toolsToAutoPatch":["vite","jest","webpack","next.js","cypress","http-server","serve","live-server","nuxt","remix","qwik","hydrogen","serverless","astro","live-server-extension","live-preview-extension"],"showErrors":true,"showLogsOnlyFromOpenedFiles":false,"showLogsFromLogPointsOnly":false,"showLogsFromMultipleRunningToolsTogether":false},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","boss","192.168.0.246","192.168.88.1","192.168.118.1"],"host":"127.0.0.1"}
08:21:49.406 info host starting tools discovery and integration
08:21:49.411 info host looking for tools to patch
08:21:49.413 info host worker Starting Host Worker...
08:21:49.424 info host found 1 tool file(s) to patch
08:21:49.424 info host patching d:\IDE\VSCodeForStream\data\extensions\ritwickdey.liveserver-5.7.9\node_modules\live-server\index.js
08:21:49.491 info host completed tools discovery and integration

```


![image](https://github.com/wallabyjs/console-ninja/assets/8464662/80693d06-6ae2-4bde-9ce7-7eda4d477bad)

**Comment by ArtemGovorov** [comment]
Thanks, can you please run through the updated set of steps below and share the result:

**Comment by ArtemGovorov** [comment]
Thanks, can you please run through the updated set of steps below and share the result:
- open Ninja trace.log file, clear and save it,
- close all instances of VS Code, open an instance of VS Code with your project,
- if Console Ninja is paused, **start Console Ninja before starting Live Server** (so that the `paused` is changed to `inactive `  icon)
- start Live Server on the project,
- copy and content of the Ninja trace.log and paste it here in response.

**Comment by ArtemGovorov** [comment]
@Alecto @alia104 This time from the log it looks like it has got a bit further and should now be working. Please try:
- editing your code (where `console.log` is located),
- and also try refreshing the web page in browser. 

If it doesn't help, please create a sample GitHub repo demonstrating the issue, we are happy to take a look.

**Comment by Alecto** [comment]
I checked and did what you wrote. No, the plugin does not work. All the same: "Console Ninja is inactive". Thanks for taking the time and trying to help, but apparently it's not fate to use this plugin.

**Comment by alia104** [comment]
My issue seems intermittent. Whenever I see the console ninja symbol at the top of my IDE I know it's working. Even when I run through all the steps to ensure Live Server is connected it still may not work.

**Comment by smcenlly** [comment]
> There's a chance the problem could be related to being on a VPN

Thanks for letting us know. We have definitely encountered VPN issues with our other product(s), [Wallaby.js](https://wallabyjs.com) and [Quokka.js](https://quokkajs.com), which work in a similar way to Console Ninja.

Console Ninja communicates on your local machine using a localhost WebSocket. We have seen that some VPN providers have experimental settings that will block this traffic. Since everything is localhost, there's no security concern.

**Comment by zome5** [comment]
I've had the same issue. I fixed it by changing the `Feature Set` in settings to `Community`
![ninja-console](https://github.com/wallabyjs/console-ninja/assets/128717256/7c39a79d-ecf3-4e99-8d73-e116946937a8)
By default it is set to `Premium`. Immediately after changing it status of the extension switched to `running`.

**Comment by smcenlly** [comment]
In terms of how Console Ninja works, there's no technical difference between the `Pro` and `Community` edition that would stop Console Ninja from working.

We expect you may have set your Console Ninja configuration options in such a way that you are not getting the output you are expecting. Please try setting your Console Ninja configuration to the defaults, I think then `Pro` will work for you.

If it's still not working for you, we'd appreciate you providing your Console Ninja log file from the `~/.console-ninja/trace.log` file in your user home directory

<img width="1136" alt="image" src="https://github.com/wallabyjs/console-ninja/assets/2074089/916653d4-c36f-4847-87f2-f3a9d5149b4b">

**Comment by zome5** [comment]
Now I cannot make it run, no matter which Feature Set I choose. 
Log: 
`
15:47:13.769 info extension stopping host
15:47:13.774 info host removing cache folder: 31b9f7957b1bf817d312a4797c61fa64
15:47:21.662 info buildHook-32864 installing build hook for live-server-extension at d:\program files (x86)\microsoft vs code, node v16.17.1
15:47:27.900 info extension stopping host
15:47:28.227 info extension host process output: port:[50802]
15:47:28.203 info host ------------ starting ------------
15:47:28.221 info host removing cache folder: 975fda819e7aeaf14a560049baa5ff9b
15:47:28.227 info extension 
15:47:28.242 info host client connected: extension { address: '::', family: 'IPv6', port: 50802 }
15:47:28.351 info host starting host: {"extensionFolder":"c:\\Users\\Alienware\\.vscode\\extensions\\wallabyjs.console-ninja-0.0.206","runtimeHookFile":"c:\\Users\\Alienware\\.vscode\\extensions\\wallabyjs.console-ninja-0.0.206\\out\\runtimeHook\\tracer.js","globalErrorHandlerFile":"c:\\Users\\Alienware\\.vscode\\extensions\\wallabyjs.console-ninja-0.0.206\\out\\runtimeHook\\errorHandler.js","filesToInstrument":[{"path":"d:\\programowanie\\vscode\\trening\\sierpień\\drag und drop\\chess\\movepatterns.js","reason":"search"},{"path":"d:\\programowanie\\VSCode\\trening\\Sierpień\\drag und drop\\chess\\main.js","reason":"open"}],"debug":false,"config":{"dateTimeFormat":"HH:mm:ss.SSS","showDateTime":true,"installBinToPath":true,"openLinksInEditor":true,"maxLogViewerEntries":15000,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":false,"searchUrl":"https://www.google.com/search?q={query}","enableProFeatures":false,"toolsToAutoPatch":["vite","jest","webpack","next.js","cypress","http-server","serve","live-server","nuxt","remix","qwik","hydrogen","serverless","astro","live-server-extension","live-preview-extension"],"showErrors":true,"showLogsOnlyFromOpenedFiles":false,"showLogsFromLogPointsOnly":false,"showLogsFromMultipleRunningToolsTogether":false},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","DESKTOP-T287QB4","192.168.178.56"],"host":"127.0.0.1"}
15:47:28.366 info host starting tools discovery and integration
15:47:28.369 info host worker Starting Host Worker...
15:47:28.372 info host looking for tools to patch
15:47:28.388 info host found 1 tool file(s) to patch
15:47:28.389 info host patching c:\Users\Alienware\.vscode\extensions\ritwickdey.liveserver-5.7.9\node_modules\live-server\index.js
15:47:28.396 info host completed tools discovery and integration
15:47:28.742 info buildHook-32864 allowed tools: vite, jest, webpack, next.js, cypress, http-server, serve, live-server, nuxt, remix, qwik, hydrogen, serverless, astro, live-server-extension, live-preview-extension, node
15:47:28.742 info buildHook-32864 running tools: live-server-extension
15:47:28.745 info buildHook-32864 installing fs interceptor
15:47:28.748 info host client connected: buildHook { address: '::', family: 'IPv6', port: 50802 }
15:47:28.749 info buildHook-32864 allowed tools: vite, jest, webpack, next.js, cypress, http-server, serve, live-server, nuxt, remix, qwik, hydrogen, serverless, astro, live-server-extension, live-preview-extension, node
15:47:28.749 info buildHook-32864 running tools: live-server-extension`

**Comment by Alecto** [comment]
I tried to set pro or community - it doesnt work

**Comment by smcenlly** [comment]
> Now I cannot make it run, no matter which Feature Set I choose.

@zome5 - your log file indicates that Console Ninja did not detect that Live Server has started. The most likely reason for this is that Console Ninja is **Paused**. You should be able to fix this by running the VS Code command `Console Ninja: Start`. If the command **is not available**, it means that Console Ninja is already started. If the command **is available** then it likely explains the cause of your problem.

> I tried to set pro or community - it doesnt work

@Alecto - We don't believe that Console Ninja working/not working will be affected by this setting. Can you please confirm that you are not behind a VPN / Firewall as that may be causing a problem. We're also wondering if Console Ninja works for you when not using Live Server? For example if you use a [`create react app`](https://create-react-app.dev/docs/getting-started) project?

**Comment by tsdineshjai** [comment]
Lol. After following all steps that has been mentioned , its still no use of using this extension.

**Comment by smcenlly** [comment]
@tsdineshjai - can you please:

- open Ninja `~/.console-ninja/trace.log` file, clear and save it,
- close all instances of VS Code, open an instance of VS Code with your project,
- if Console Ninja is paused, start Console Ninja before starting Live Server (so that the paused is changed to inactive icon)
- start Live Server on the project,
- copy and content of the Ninja `~/.console-ninja/trace.log` and paste it in your response.

**Comment by vilelacode** [comment]
Don't work :( 

17:34:50.361 info host ------------ starting ------------
17:34:50.447 info host client connected: extension { address: '127.0.0.1', family: 'IPv4', port: 60595 }
17:34:50.657 info host worker Starting Host Worker...
17:34:50.813 info host starting host: {"extensionFolder":"c:\\Users\\rhcp_\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.324","runtimeHookFile":"c:\\Users\\rhcp_\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.324\\out\\runtimeHook\\tracer.js","globalErrorHandlerFile":"c:\\Users\\rhcp_\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.324\\out\\runtimeHook\\errorHandler.js","networkLoggingHandlerFile":"c:\\Users\\rhcp_\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.324\\out\\runtimeHook\\networkLoggingHandler.js","filesToInstrument":[{"path":"c:\\users\\rhcp_\\onedrive\\área de trabalho\\js aulas\\projetos\\01\\js\\script.js","reason":"search"},{"path":"c:\\Users\\rhcp_\\OneDrive\\Área de Trabalho\\JS Aulas\\projetos\\01\\index.html","reason":"open"},{"path":"c:\\Users\\rhcp_\\OneDrive\\Área de Trabalho\\JS Aulas\\projetos\\01\\js\\script.js","reason":"open"}],"debug":false,"config":{"dateTimeFormat":"HH:mm:ss.SSS","dateTimeDisplayMode":"Absolute","installBinToPath":true,"openLinksInEditor":true,"maxLogViewerEntries":15000,"showWelcomeMessageInTools":true,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":false,"searchUrl":"https://www.google.com/search?q={query}","enableProFeatures":true,"toolsToAutoPatch":["vite","angular","jest","webpack","next.js","nest.js","cypress","http-server","serve","live-server","nuxt","remix","qwik","hydrogen","serverless","astro","live-server-extension","live-preview-extension"],"showErrors":true,"showLogsOnlyFromOpenedFiles":true,"showLogsFromLogPointsOnly":true,"showLogsFromMultipleRunningToolsTogether":true,"predictiveLoggingEnabled":true,"allowConsoleLogToTriggerPredictions":true,"showNetworkRequests":true},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","Viles","192.168.0.74","192.168.213.1","192.168.182.1"],"host":"127.0.0.1","forcedStart":true,"startTime":1718483689897}
17:34:50.859 info host starting tools discovery and integration
17:34:50.871 info host installing PATH C:\Users\rhcp_\.console-ninja\.bin
17:34:53.943 info host looking for tools to patch
17:34:53.968 info host found 1 tool file(s) to patch
17:34:53.970 info host patching c:\Users\rhcp_\.vscode\extensions\ritwickdey.liveserver-5.7.9\node_modules\live-server\index.js
17:34:53.978 info host completed tools discovery and integration
17:34:53.983 ERR! host uncaught exception: No clients connected

**Comment by smcenlly** [comment]
@vilelacode - can you please provide us with a sample repo where it's not working?

**Comment by MaxSmirnov01** [comment]
The same problem. VPN is turned off.

09:11:08.335 info host ------------ starting ------------
09:11:08.393 info extension host process output: port:[63005]
09:11:08.393 info extension 
09:11:08.415 info host client connected: extension { address: '127.0.0.1', family: 'IPv4', port: 63005 }
09:11:08.563 info host worker Starting Host Worker...
09:11:08.806 info host starting host: {"extensionFolder":"c:\\Users\\acer\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.327","runtimeHookFile":"c:\\Users\\acer\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.327\\out\\runtimeHook\\tracer.js","globalErrorHandlerFile":"c:\\Users\\acer\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.327\\out\\runtimeHook\\errorHandler.js","networkLoggingHandlerFile":"c:\\Users\\acer\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.327\\out\\runtimeHook\\networkLoggingHandler.js","filesToInstrument":[{"path":"\\\\wsl.localhost\\ubuntu\\home\\max-smirnov-acer-aspire7\\smartway\\src\\store\\store.ts","reason":"search"},{"path":"\\\\wsl.localhost\\ubuntu\\home\\max-smirnov-acer-aspire7\\smartway\\src\\components\\layout\\input\\input.tsx","reason":"search"},{"path":"\\\\wsl.localhost\\Ubuntu\\home\\max-smirnov-acer-aspire7\\smartway\\src\\components\\layout\\Input\\Input.tsx","reason":"open"}],"debug":false,"config":{"dateTimeFormat":"HH:mm:ss.SSS","dateTimeDisplayMode":"Absolute","installBinToPath":true,"openLinksInEditor":true,"maxLogViewerEntries":15000,"showWelcomeMessageInTools":true,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":false,"searchUrl":"https://www.google.com/search?q={query}","enableProFeatures":true,"toolsToAutoPatch":["vite","angular","jest","webpack","next.js","nest.js","cypress","http-server","serve","live-server","nuxt","remix","qwik","hydrogen","serverless","astro","live-server-extension","live-preview-extension"],"showErrors":true,"showLogsOnlyFromOpenedFiles":false,"showLogsFromLogPointsOnly":false,"showLogsFromMultipleRunningToolsTogether":false,"predictiveLoggingEnabled":true,"allowConsoleLogToTriggerPredictions":false,"showNetworkRequests":true},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","LAPTOP-S6IITOTF","172.25.48.1","192.168.0.133"],"host":"127.0.0.1","forcedStart":true,"startTime":1720419068013}
09:11:08.832 info host starting tools discovery and integration
09:11:08.870 info host looking for tools to patch
09:11:09.296 info host found 2 tool file(s) to patch
09:11:09.297 info host patching c:\Users\acer\.vscode\extensions\ritwickdey.liveserver-5.7.9\node_modules\live-server\index.js
09:11:09.307 info host patching \\wsl.localhost\ubuntu\home\max-smirnov-acer-aspire7\smartway\node_modules\vite\bin\vite.js
09:11:09.330 info host Host's startup and patch time: 1317 ms
09:11:09.330 info host completed tools discovery and integration
09:11:09.336 ERR! host uncaught exception: No clients connected

**Comment by KevinMCruzP** [comment]
@vilelacode try to delete the .console-ninja directory with the command "rm -rf ~/.console-ninja" or directly in the root, it depends on what OS you are using.
 
![console-ninja](https://github.com/wallabyjs/console-ninja/assets/72741197/bc31f154-8d83-4825-bace-c0ea2161809d)

So it should work already, but if it doesn't work yet, try Reload Window with the shortcut "CTRL + Shift + P" and writing "Reload Window"

![image](https://github.com/wallabyjs/console-ninja/assets/72741197/7a320c3b-9311-4806-b05b-8af95e7f3feb)

I just do it now, and it works, but I'm not sure if it is a definitive solution, probably not.

**Comment by smcenlly** [comment]
@MaxSmirnov01 - can you please provide us with:

1. A sample repo where it's not working for you
2. The command you're using to start your application

**Comment by MaxSmirnov01** [comment]
@smcenlly It doesn't work in any project.

1. https://github.com/MaxSmirnov01/image-generator-ts
2. npm run dev

**Comment by smcenlly** [comment]
@MaxSmirnov01 - your project is working for us. Re-reviewing your log, I can see you're using WSL. It looks like you're opening your WSL project using the network drive share:

```
path: \\wsl.localhost\ubuntu\home\...
```

This is almost certainly the cause of your problem.

Console Ninja supports WSL when you open your project in VS Code using [Microsoft's WSL extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) and [follow their development workflow](https://code.visualstudio.com/docs/remote/wsl). If you open your project using Microsoft's WSL extension and follow the recommended development workflow, it should work for you.

**Comment by aliyanissohaib** [comment]
i am facing the same error in react project ninja is inactive but in terminal it is showing serving running

00:54:04.703 info extension stopping host
00:54:04.744 info host removing cache folder: fc6dc5f4273f423e4c47088215e6c303
00:54:16.688 info extension stopping host
00:54:17.757 info host ------------ starting ------------
00:54:17.984 info extension host process output: port:[51387]
00:54:17.984 info extension 
00:54:18.027 info host client connected: extension { address: '127.0.0.1', family: 'IPv4', port: 51387 }
00:54:18.357 info host worker Starting Host Worker...
00:54:18.546 info host starting host: {"extensionFolder":"c:\\Users\\ALIYAN\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.340","runtimeHookFile":"c:\\Users\\ALIYAN\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.340\\out\\runtimeHook\\tracer.js","globalErrorHandlerFile":"c:\\Users\\ALIYAN\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.340\\out\\runtimeHook\\errorHandler.js","networkLoggingHandlerFile":"c:\\Users\\ALIYAN\\.vscode\\extensions\\wallabyjs.console-ninja-1.0.340\\out\\runtimeHook\\networkLoggingHandler.js","filesToInstrument":[{"path":"d:\\hassan proj\\final\\api\\controllers\\user.controller.js","reason":"search"},{"path":"d:\\hassan proj\\final\\api\\controllers\\test.controller.js","reason":"search"},{"path":"d:\\hassan proj\\final\\api\\controllers\\post.controller.js","reason":"search"},{"path":"d:\\hassan proj\\final\\api\\controllers\\message.controller.js","reason":"search"},{"path":"d:\\hassan proj\\final\\api\\controllers\\chat.controller.js","reason":"search"},{"path":"d:\\hassan proj\\final\\api\\controllers\\auth.controller.js","reason":"search"},{"path":"d:\\hassan proj\\final\\api\\app.js","reason":"open"}],"debug":false,"config":{"dateTimeFormat":"HH:mm:ss.SSS","dateTimeDisplayMode":"Absolute","installBinToPath":true,"openLinksInEditor":true,"maxLogViewerEntries":15000,"showWelcomeMessageInTools":true,"fontSize":10,"outputMode":"In View","captureFunctions":false,"searchUrl":"https://www.google.com/search?q={query}","enableProFeatures":false,"toolsToAutoPatch":["vite","angular","jest","webpack","next.js","nest.js","cypress","http-server","serve","live-server","nuxt","remix","qwik","hydrogen","serverless","astro","live-server-extension","live-preview-extension"],"showErrors":true,"showLogsOnlyFromOpenedFiles":false,"showLogsFromLogPointsOnly":false,"showLogsFromMultipleRunningToolsTogether":false,"predictiveLoggingEnabled":true,"allowConsoleLogToTriggerPredictions":true,"showNetworkRequests":true},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","DESKTOP-ODJJV6L","192.168.0.103"],"host":"127.0.0.1","forcedStart":true,"startTime":1723492456720}
00:54:18.605 info host starting tools discovery and integration
00:54:18.622 info host looking for tools to patch
00:54:18.807 info host found 2 tool file(s) to patch
00:54:18.810 info host patching c:\Users\ALIYAN\.vscode\extensions\ritwickdey.liveserver-5.7.9\node_modules\live-server\index.js
00:54:18.812 info host patching c:\Users\ALIYAN\.vscode\extensions\ms-vscode.live-server-0.4.14\out\extension.js
00:54:18.933 info host completed tools discovery and integration
00:54:18.947 ERR! host uncaught exception: No clients connected
00:54:29.247 info buildHook-204 installing build hook for node at d:\hassan proj\final\api, node v20.11.1
00:54:29.250 ERR! buildHook-204 failed to install build hook: Cannot read properties of undefined (reading 'substring')


![Untitled](https://github.com/user-attachments/assets/76d91463-fe24-4aac-88f6-6f3c857164ed)

**Comment by smcenlly** [comment]
@aliyanissohaib - thanks for reporting the issue. This is fixed in the latest version of Console Ninja, `v1.0.341`.

**Comment by rosscondie** [comment]
I came here looking for answers as I wasn't able to get logging to work. I was using WSL. 

I did not receive any error message within the WSL terminal but when I switched to Powershell out popped a message letting me know "Using vite v5.4.2 is not supported yet in community edition of Console Ninja.".

Tried without vite in a simple project and all works fine. Just in case anyone else is using this version of vite. 

🫡

**Comment by Lemonwsl** [comment]
> "rm -rf ~/.console-ninja"
Fixed my issue
🫡

**Comment by ZakiRaihan4636** [comment]
Fixed Issue: Removed Console Ninja configuration directory

Resolved the issue by executing rm -rf ~/.console-ninja, which cleared out the problematic configuration and restored normal functionality. Thanks for the help!

**Comment by smcenlly** [comment]
@TheArgentinian - we'd be happy to help. Please [create a new issue[(https://github.com/wallabyjs/console-ninja/issues/new?assignees=&labels=bug%2Crequires+triage&projects=&template=bug.yml&title=%5BBug%5D%3A+) and provide the content requested in the issue template. This should give us enough information to understand your configuration / setup and start troubleshooting.

**Comment by m-shrif** [comment]
ninja is  inactive we try many time but not work for inline logs at least  we run these command 
Remove-Item -Recurse -Force C:\Users\Sharif\.console-ninja
after this commad issue resolve successfully.

**Comment by addrissio** [comment]
✘ next.js v15.2.3 is not yet supported in the Community edition of Console Ninja.

**Comment by behrangsa** [comment]
I am also having th same problem with Astro 6:

Here are the logs:

```
23:56:53.057 info extension stopping host
23:56:53.083 info host removing cache folder: 6db27a91bb31467799992e54deff3729
23:56:53.087 info extension Console Ninja extension version: 1.0.517
23:56:53.126 info extension client disconnected: 1006 
23:56:53.374 info host ------------ starting ------------
23:56:53.428 info host client connected: extension { address: '127.0.0.1', family: 'IPv4', port: 37067 }
23:56:53.448 info host starting host: {"extensionFolder":"/home/behrang/.vscode/extensions/wallabyjs.console-ninja-1.0.517","runtimeHookFile":"/home/behrang/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/runtimeHook/tracer.js","globalErrorHandlerFile":"/home/behrang/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/runtimeHook/errorHandler.js","networkLoggingHandlerFile":"/home/behrang/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/runtimeHook/networkLoggingHandler.js","filesToInstrument":[{"path":"/home/behrang/Code/behrangsa/behrang.org/src/components/Desktop.astro","reason":"open"},{"path":"/home/behrang/Code/behrangsa/behrang.org/src/layouts/Layout.astro","reason":"open"},{"path":"/home/behrang/Code/behrangsa/behrang.org/src/pages/index.astro","reason":"open"},{"path":"/home/behrang/Code/behrangsa/behrang.org/astro.config.mjs","reason":"open"}],"debug":false,"config":{"dateTimeFormat":"HH:mm:ss.SSS","dateTimeDisplayMode":"Absolute","installBinToPath":false,"openLinksInEditor":true,"maxLogViewerEntries":15000,"showWelcomeMessageInTools":true,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":true,"enableProFeatures":false,"toolsToAutoPatch":["vite","http-server","serve","live-server","astro","live-server-extension","live-preview-extension"],"showErrors":true,"showLogsOnlyFromOpenedFiles":false,"showLogsFromLogPointsOnly":false,"showLogsFromMultipleRunningToolsTogether":false,"predictiveLoggingEnabled":false,"allowConsoleLogToTriggerPredictions":false,"showNetworkRequests":false,"logCaptureOptions":{"resolveGetters":false,"defaultLimits":{"props":100,"elements":100,"strLength":51200,"totalStrLength":51200,"autoExpandLimit":5000,"autoExpandMaxDepth":10},"reducedLimits":{"props":5,"elements":5,"strLength":256,"totalStrLength":768,"autoExpandLimit":30,"autoExpandMaxDepth":2},"reducePolicy":{"perLogpoint":{"reduceOnCount":50,"reduceOnAccumulatedProcessingTimeMs":100,"resetWhenQuietMs":500,"resetOnProcessingTimeAverageMs":100},"global":{"reduceOnCount":1000,"reduceOnAccumulatedProcessingTimeMs":300,"resetWhenQuietMs":50,"resetOnProcessingTimeAverageMs":100}}}},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","10.0.2.2","nixos","192.168.1.105"],"host":"127.0.0.1","forcedStart":true,"startTime":1773273413084}
23:56:53.471 info host starting tools discovery and integration
23:56:53.477 info host looking for tools to patch
23:56:53.595 info host worker Starting Host Worker...
23:56:54.742 info host found 3 tool file(s) to patch
23:56:54.743 info host patching /home/behrang/Code/behrangsa/behrang.org/node_modules/.pnpm/node_modules/vite/bin/vite.js
23:56:54.797 info host Host's startup and patch time: 1713 ms
23:56:54.797 info host completed tools discovery and integration
23:56:54.810 WARN host No clients connected
```

**Comment by behrangsa** [comment]
FYI, fish shell is not also supported:

```
source ~/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/shell/console-ni
nja
~/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/shell/console-ninja (line 3): $@ is not supported. In fish, please use $argv.
$@
 ^
from sourcing file ~/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/shell/console-ninja
source: Error while reading file “/home/amadeus/.vscode/extensions/wallabyjs.console-ninja-1.0.517/out/shell/console-ninja”
```

**Comment by NikGovorov** [comment]
@behrangsa Thanks for reporting the issues. We've addressed them in v1.0.518.
